src/routes/user/userComponents/userPlanData.svelte

```diff
@@ -1,14 +1,151 @@
+        hideDismiss: true, // Hide the dismiss button on toast
+        timeout: 3000, // Auto dismiss toast after 3 seconds
+        background: "variant-filled-warning",
+    };
+
+    // Error toast
+    const errorToast = {
+        message: "Deine Daten konnten nicht aktualisiert werden",
+        hideDismiss: true, // Hide the dismiss button on toast
+        timeout: 3000, // Auto dismiss toast after 3 seconds
+        background: "variant-filled-error",
+    };
+
+    // Functions
+    function checkPlanNameDidntChange(planValue) {
+        return planValue == lastplanValue;
+    }
+
+    function resolvePlanNameToPlanId(planName) {
+        for (let index = 0; index < planData.length; index++) {
+            if (planData[index].name == planName) {
+                return planData[index].id;
+            }
+        }
+        throw new Error("Planname is not in backend plan list");
+    }
+
+    // Retrieves all available plans at mounting time
+    onMount(() => {
+        planData = new Promise(async (resolve, reject) => {
+            try {
+                // Fetch backend for plan data with retry policy
+                const response = await tryFetchingRestricted(urls.get.allPlans);
+
+                if (response.ok) {
+                    planData = await response.json();
+                    lastplanValue = resolvePlanNameToPlanId(
+                        personalData.planName,
+                    );
+                } else {
+                    throw new Error("Error while fetching data");
+                }
+            } catch (err) {
+                reject(err); // Reject the promise so Svelte can handle it
+            }
+        });
+    });
 
-    // ToDo maybe show possible plan and option to change current one
+    // Makes a request to the backend to update the plan data of the user
+    async function updatePlanData() {
+        if (formEditEnabled) {
+            formEditEnabled = false;
+        }
+        if (checkPlanNameDidntChange(planReference.value)) {
+            toastStore.trigger(warningToast);
+            return;
+        }
+        try {
+            const mybody = { planId: planReference.value };
+            console.log(mybody);
+            const response = await tryFetchingRestricted(
+                urls.put.newPlanData,
+                "PUT",
+                mybody,
+            );
+            if (response.ok) {
+                toastStore.trigger(successToast);
+                lastplanValue = planReference.value;
+            } else {
+                throw new Error("Update failed");
+            }
+        } catch (err) {
+            console.log(err);
+            toastStore.trigger(errorToast);
+        }
+    }
 </script>
 
-<div>
-    Aktueller Tarif
-    <!-- Currently a placeholder; will be changed when backend fetch possible -->
-    {userID}
-    <button class="btn text-warning-500">Tarif ändern</button>
-</div>
+{#await planData}
+    <p>loading</p>
```

Comment:
```suggestion
    <p>Tarife werden geladen...</p>
```
Irgendwie so, feel free to change